From a463b7b9a465cd5f1ff8df32f863dd0161a5f27d Mon Sep 17 00:00:00 2001
From: Tsuyoshi Ozawa <ozawa@apache.org>
Date: Wed, 2 Sep 2015 02:01:51 +0900
Subject: [PATCH 1191/2748] HADOOP-10365. BufferedOutputStream in
 FileUtil#unpackEntries() should be closed in
 finally block. Contributed by Kiran Kumar M R and
 Sanghyun Yun.

(cherry picked from commit dd149adeace8727864371c5a1484c6534f8b450b)
(cherry picked from commit 6078172fee6b0843cf4aa0fce5090d728ab7e53a)

Change-Id: Ibb1f0582ad7fc5f1c16deb533218f3c233920ea8
---
 .../main/java/org/apache/hadoop/fs/FileUtil.java   |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileUtil.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileUtil.java
index ff22264..8e6a99f 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileUtil.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileUtil.java
@@ -732,15 +732,15 @@ private static void unpackEntries(TarArchiveInputStream tis,
 
     int count;
     byte data[] = new byte[2048];
-    BufferedOutputStream outputStream = new BufferedOutputStream(
-        new FileOutputStream(outputFile));
+    try (BufferedOutputStream outputStream = new BufferedOutputStream(
+        new FileOutputStream(outputFile));) {
 
-    while ((count = tis.read(data)) != -1) {
-      outputStream.write(data, 0, count);
-    }
+      while ((count = tis.read(data)) != -1) {
+        outputStream.write(data, 0, count);
+      }
 
-    outputStream.flush();
-    outputStream.close();
+      outputStream.flush();
+    }
   }
   
   /**
-- 
1.7.9.5

